File: ios/Tradeguruelectrical/Views/Blocks/RegulationView.swift
import SwiftUI

struct RegulationView: View {
    let code: String?
    let clause: String?
    let summary: String?

    var body: some View {
        HStack(spacing: 0) {
            Rectangle()
                .fill(Color.modeResearch)
                .frame(width: 4)

            VStack(alignment: .leading, spacing: 6) {
                if let code {
                    Text(code)
                        .font(.system(size: 15, weight: .bold))
                        .foregroundStyle(Color.tradeText)
                }

                if let clause {
                    Text(clause)
                        .font(.system(size: 13))
                        .foregroundStyle(Color.tradeTextSecondary)
                }

                if let summary {
                    Text(summary)
                        .font(.system(size: 14))
                        .foregroundStyle(Color.tradeText)
                }
            }
            .padding(12)
            .frame(maxWidth: .infinity, alignment: .leading)
        }
        .background(Color.tradeSurface)
        .clipShape(.rect(cornerRadius: 12))
    }
}

File: android/app/src/main/java/com/tradeguru/electrical/ui/views/blocks/RegulationView.kt
package com.tradeguru.electrical.ui.views.blocks

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.IntrinsicSize
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.tradeguru.electrical.ui.theme.LocalTradeGuruColors

@Composable
fun RegulationView(code: String?, clause: String?, summary: String?) {
    val colors = LocalTradeGuruColors.current

    Row(
        modifier = Modifier
            .fillMaxWidth()
            .height(IntrinsicSize.Min)
            .clip(RoundedCornerShape(12.dp))
            .background(colors.tradeSurface)
    ) {
        Box(
            modifier = Modifier
                .width(4.dp)
                .fillMaxHeight()
                .background(colors.modeResearch)
        )

        Column(
            verticalArrangement = Arrangement.spacedBy(6.dp),
            modifier = Modifier
                .fillMaxWidth()
                .padding(start = 16.dp, end = 12.dp, top = 12.dp, bottom = 12.dp)
        ) {
            if (code != null) {
                Text(
                    text = code,
                    style = TextStyle(
                        fontSize = 15.sp,
                        fontWeight = FontWeight.Bold,
                        color = colors.tradeText
                    )
                )
            }

            if (clause != null) {
                Text(
                    text = clause,
                    style = TextStyle(
                        fontSize = 13.sp,
                        color = colors.tradeTextSecondary
                    )
                )
            }

            if (summary != null) {
                Text(
                    text = summary,
                    style = TextStyle(
                        fontSize = 14.sp,
                        color = colors.tradeText
                    )
                )
            }
        }
    }
}
